persistent/Database/Persist/Sql/Orphan/PersistUnique.hs

```diff
@@ -18,43 +21,75 @@ import qualified Data.Conduit.List as CL
 import Control.Monad.Trans.Reader (ask, withReaderT)
 import Control.Monad (when, liftM)
 
-defaultUpsert :: (MonadIO m, PersistEntity record, PersistUniqueWrite backend
-                 , PersistEntityBackend record ~ BaseBackend backend) 
-                => record -> [Update record] -> ReaderT backend m (Entity record)
+defaultUpsert
+    :: (MonadIO m
+       ,PersistEntity record
+       ,PersistUniqueWrite backend
+       ,PersistEntityBackend record ~ BaseBackend backend)
+    => record -> [Update record] -> ReaderT backend m (Entity record)
 defaultUpsert record updates = do
-  uniqueKey <- onlyUnique record
-  upsertBy uniqueKey record updates
+    uniqueKey <- onlyUnique record
+    upsertBy uniqueKey record updates
 
-instance PersistUniqueWrite SqlBackend where
+escape :: DBName -> T.Text
+escape (DBName s) = T.pack $ '"' : go (T.unpack s) ++ "\""
+  where
+    go "" = ""
+    go ('"':xs) = "\"\"" ++ go xs
+    go (x:xs) = x : go xs
```

Comment:
Thanks for the feedback, I will change it to `escapeQuote`. Actually, I didn't write the `escape` function itself. I took it from some other parts in `persistent`.